src/editors/application/analysis-features/enabled-analysis-features-repository.php

```diff
@@ -44,12 +36,32 @@ public function __construct( Analysis_Feature_Interface ...$plugin_features ) {
 	 * @return Analysis_Features_List The analysis list.
 	 */
 	public function get_enabled_features(): Analysis_Features_List {
-		if ( \count( $this->enabled_analysis_features->parse_to_legacy_array() ) === 0 ) {
-			foreach ( $this->plugin_features as $plugin_feature ) {
+		$enabled_analysis_features = new Analysis_Features_List();
```

Comment:
Why refactor this? It increases the scope of the PR without much benefit and I'd rather we stayed focused on the task at hand

Reply:
Basically, why don't we keep this as is and create the `get_enabled_features_by_keys()` function in the same spirit?

Reply:
I needed to refactor this since for some reason I added the analysis features list as a class property and that way it cant be resued in both methods which was weird.